core/domain/protofiles/training_job_data.proto

```diff
@@ -0,0 +1,33 @@
+// coding: utf-8
+//
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "core/domain/protofiles/text_classifier.proto";
+
+message TrainingJobData {
+  message JobResult {
+    string job_id = 1;
```

Comment:
For this & other IDs: consider wrapping the ID in an opaque object to provide type safety, e.g.:

```proto
message JobId {
  // The internal ID value. This should not be used directly.
  string internal_job_id = 1;
}
```

This forces downstream code to use 'JobId' everywhere, which provides clearer contracts in your classes and reduces the potential for developer error in the future.

Reply:
However, to get and set the exact ID we will still have to call the `internal_job_id`. Do you mean to suggest that we have a corresponding python domain object to this necessary conversion as well?

Reply:
Need more clarification on this.

Reply:
You're right that we still need to set the internal ID, but we would only do that in the context of whatever object that's privileged to interact with that internal element (such as a JobId domain object). I think reflecting the state via domain objects is fine (though maybe unnecessary--JobId will be an object generated for the proto & can be shared itself).

Reply:
I see, however, proto is only generated by functions which convert domain object to proto messages. Otherwise, we are not using proto for internal communication of job data, that still happens through domain objects. In such case, is it still necessary to have a wrapper over JobId? Since, in the Porto, job_id is only set by specific converter functions (which convert internal job data domain object to proto).

Reply:
Hmm that’s a fair point. If job ID itself will never be passed around then I don’t see value in wrapping it in its own proto.